### PR TITLE
Add signup shifts query by posting id

### LINKED ISSUE #190: Signup endpoint for Admin shift calendar
For the shift calendar, we want to get signups by posting id and by month.

A new `getShiftSignupsForPostingInDateRange` method is created in `ShiftSignupService`. It should accept `postingID: ID!`, `dateRange: {min: Date, max: Date}` parameters.

### PR DESCRIPTION
**Pull request by MatoPlus**
## Ticket link

Closes #190 

## Implementation description
* Add simple query to get signup shifts by posting ids

## Steps to test
1. create signup shifts of posting
2. call query to get created queries

## What should reviewers focus on?
* correctness

## Checklist
- [X] My PR name is descriptive and in imperative tense
- [X] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [X] I have run the appropriate linter(s)
- [X] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/services/implementations/shiftSignupService.ts [hunk 1]
```diff
@@ -62,6 +62,48 @@ class ShiftSignupService implements IShiftSignupService {
...
+    try {
+      const filter: Prisma.SignupWhereInput = {
+        AND: [
+          {
+            shift: {
+              postingId: Number(postingId),
+              startTime: {
+                gte: startDate,
+                lte: endDate,
+              },
```
Comment (MatoPlus) [inline backend/services/implementations/shiftSignupService.ts:80]: @uwblueprint/sistering-w22-pls Not 100% sure if this filter meets the requirement. I understood this as to get all signups that has started within a time range.

Comment (MatoPlus) [inline backend/services/implementations/shiftSignupService.ts:80]: @uwblueprint/sistering-w22-pls Not 100% sure if this filter meets the requirement. I understood this as to get all signups that has started within a time range.

File: backend/services/implementations/shiftSignupService.ts [hunk 2]
```diff
@@ -62,6 +62,48 @@ class ShiftSignupService implements IShiftSignupService {
...
+            },
+          },
+          signupStatus
+            ? {
+                status: signupStatus,
+              }
+            : {},
+        ],
+      };
+      const shiftSignups = await prisma.signup.findMany({
```
Comment (MatoPlus) [inline backend/services/implementations/shiftSignupService.ts:90]: I'm not getting anything from this prisma call atm. I'll need to investigate why.

File: backend/services/interfaces/shiftSignupService.ts [hunk 3]
```diff
@@ -40,6 +40,21 @@ interface IShiftSignupService {
...
+   * @param startDate the start range to filter
+   * @param endDate the end range to filter
+   * @returns an array of ShiftSignupResponseDTOs for each shift within the date range
+   * @throws Error if the shift signup retrieval fails
+   */
+  getShiftSignupsForPostingInDateRange(
+    postingId: string,
+    startDate: Date,
+    endDate: Date,
+    signupStatus: SignupStatus | null,
```
Comment (MatoPlus) [inline backend/services/interfaces/shiftSignupService.ts:56]: Not sure if this is needed...

File: backend/services/implementations/shiftSignupService.ts (hunk 1 above)
Comment (LenaNguyen) [reply backend/services/implementations/shiftSignupService.ts:80]: hm on second thought, the filtering of the shift dates should happen on the front end 😓  Similar to what you did for the volunteer availabilities table. Sorry about that!

Comment (MatoPlus) [reply backend/services/implementations/shiftSignupService.ts:80]: @LenaNguyen I'll consider that another ticket closed! I'll close the PR.

**Comment by MatoPlus** [comment]
https://github.com/uwblueprint/sistering/pull/231#discussion_r832442655

**Comment by MatoPlus** [comment]
Changing query to filter by postingId

File: backend/services/implementations/shiftSignupService.ts (hunk 1 above)
Comment (LenaNguyen) [reply backend/services/implementations/shiftSignupService.ts:80]: Yup! Whoops I didn't make that clear lol

File: backend/services/interfaces/shiftSignupService.ts (hunk 3 above)
Comment (LenaNguyen) [reply backend/services/interfaces/shiftSignupService.ts:52]: doesn't hurt to have it!

**Review by LenaNguyen — APPROVED** [review]

**Pull request merged**